ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/layered/ReadyTransactions.java

```diff
@@ -137,11 +136,24 @@ protected boolean promotionFilter(final PendingTransaction pendingTransaction) {
     return true;
   }
 
+  /**
+   * Return the full content of this layer, organized as a list of sender pending txs. For each
+   * sender the collection pending txs is ordered by nonce asc.
+   *
+   * <p>Returned sender list order detail: first the sender of the tx with the highest max gas
+   * price.
+   *
+   * @return a list of sender pending txs
+   */
   @Override
-  public Stream<PendingTransaction> stream() {
+  public synchronized List<SenderPendingTransactions> getBySender() {
```

Comment:
Actually this is a left over, removed it since not required. The synchronization of the layered txpool is all managed by the entry point `LayeredPendingTransactions` class, layers themselves are not thread safe, since they are not meant to be accessed directly, this makes it easier to reason about the synchronization